prime-dotnet-webapi/Services/BannerService.cs

```diff
@@ -42,35 +52,45 @@ public async Task<BannerViewModel> GetBannerAsync(BannerLocationCode locationCod
             return banner;
         }
 
-        public async Task<BannerDisplayViewModel> GetActiveBannerAsync(BannerLocationCode locationCode, DateTime atTime)
+        public async Task<IEnumerable<BannerViewModel>> GetBannersAsync(BannerLocationCode locationCode)
         {
-            atTime = atTime.ToUniversalTime();
-            return await _context.Banners
+            var banners = await _context.Banners
                 .Where(b => b.BannerLocationCode == locationCode)
-                .Where(b => b.StartTimestamp <= atTime && atTime <= b.EndTimestamp)
-                .ProjectTo<BannerDisplayViewModel>(_mapper.ConfigurationProvider)
-                .SingleOrDefaultAsync();
+                .ProjectTo<BannerViewModel>(_mapper.ConfigurationProvider)
+                .ToListAsync();
+
+            return banners.Select(banner =>
+            {
+                // Banner time stamps are stored as UTC, but are materialized as DateTimeKind.Unspecified
+                // Once we upgrade our EF / Npgsql, we can use NodaTime instead
+                banner.StartTimestamp = DateTime.SpecifyKind(banner.StartTimestamp, DateTimeKind.Utc);
+                banner.EndTimestamp = DateTime.SpecifyKind(banner.EndTimestamp, DateTimeKind.Utc);
+                return banner;
+            });
```

Comment:
![image](https://user-images.githubusercontent.com/39168456/124049017-71c19580-d9cc-11eb-90b0-8921e9398d9b.png)

While its not _wrong_ to use a `Select()` statement this way, one main goal of functional programming is to avoid causing side-effects. So using functional programming to intentionally cause side effects this way _feels_ bad to me. I would prefer a `foreach()` loop when you want to mutate the source objects or cause other side effects. `List` actually already has a `.ForEach()` method, so this code can be:

```C#
banners.ForEach(banner =>
{
    // Banner time stamps are stored as UTC, but are materialized as DateTimeKind.Unspecified
    // Once we upgrade our EF / Npgsql, we can use NodaTime instead
    banner.StartTimestamp = DateTime.SpecifyKind(banner.StartTimestamp, DateTimeKind.Utc);
    banner.EndTimestamp = DateTime.SpecifyKind(banner.EndTimestamp, DateTimeKind.Utc);
});

return banners;
```

Reply:
Good to know, Javascript mind is always on autopilot